frontend/src/lib/get-cluster.ts

```diff
@@ -612,6 +634,30 @@ export function getClusterStatus(
             }
 
             return { status: ccStatus, statusMessage }
+        } else if (clusterDeployment) {
```

Comment:
Seems to be there is a lot of duplication in the two if blocks and you could rework it to be simpler.  Something like

                if (
                    checkCuratorConditionFailed(CuratorCondition.curatorjob, ccConditions) &&
                    checkCuratorLatestFailedOperation(CuratorCondition.install, ccConditions)
                ) {
                    if (clusterDeployment.spec?.installed) {
                       ccStatus = ClusterStatus.posthookfailed
                    } else {
                       ccStatus = ClusterStatus.posthookfailed
                    }
                    statusMessage = clusterCurator.status?.conditions[0].message

                    return { status: ccStatus, statusMessage }
                }
            }